Realm/RLMSyncUser.mm

```diff
@@ -36,7 +36,7 @@
 
 using namespace realm;
 using ConfigMaker = std::function<Realm::Config(std::shared_ptr<SyncUser>, std::string)>;
-using PermissionGetCallback = std::function<void(std::unique_ptr<PermissionResults>, std::exception_ptr)>;
+using PermissionGetCallback = std::function<void(Results, std::exception_ptr)>;
```

Comment:
Why do we redefine this typedef rather than using `Permission::PermissionResultsCallback`?